test/common/testlib.py

```diff
@@ -571,14 +600,21 @@ def snapshot(self, title, label=None):
             self.cdp.command("clearExceptions()")
 
             filename = "{0}-{1}.png".format(label or self.label, title)
-            ret = self.cdp.invoke("Page.captureScreenshot", no_trace=True)
-            if "data" in ret:
-                with open(filename, 'wb') as f:
-                    f.write(base64.standard_b64decode(ret["data"]))
-                attach(filename)
-                print("Wrote screenshot to " + filename)
-            else:
-                print("Screenshot not available")
+            if (self.cdp.browser == "chromium"):
+                ret = self.cdp.invoke("Page.captureScreenshot", no_trace=True)
+                ret = ""
+                if "data" in ret:
+                    with open(filename, 'wb') as f:
+                        f.write(base64.standard_b64decode(ret["data"]))
+                    attach(filename)
+                    print("Wrote screenshot to " + filename)
+                else:
+                    print("Screenshot not available")
+            elif (self.cdp.browser == "firefox"):
+                # TODO: implement me
+                # Something like:
+                # Runtime.execute(':screenshot --file <path>)
```

Comment:
Is there some API like that already, and thus this is a SMOP? Or completely lacking right now? (I would think that for puppeteer support firefox must have some screenshot support?

Reply:
The API (`Page.captureScreenshot`) is missing. You can execute in your firefox console `:screenshot` and it will create screenshot. I though we could do this and use `Runtime.execute` as it is what we do normally and it behaves the same as when you execute something in console. I tried it but it did not work. I haven't yet dug deep why it is not working.

Reply:
Ack, no problem. Thus for now, debugging a failing test would need to go through attaching it to an interactive firefox.